### PR TITLE
Route with extra dashes is executed controller while it should not

### PR DESCRIPTION
**Pull request by developeruz**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14903

### CODE REVIEW

File: framework/base/Module.php [hunk 1]
```diff
@@ -649,6 +648,23 @@ public function createControllerByID($id)
...
     }
 
+    /**
+     * @param $className
+     * @param $prefix
+     * @return bool
+     */
+    private function incorrectClassNameOrPrefix($className, $prefix)
+    {
+        if (!preg_match('%^[a-z][a-z0-9\\-_]*$%', $className)) {
```
Comment (vladis84) [inline framework/base/Module.php]: ClassName can be in format CamelCase?

File: framework/base/Module.php [hunk 2]
```diff
@@ -649,6 +648,23 @@ public function createControllerByID($id)
         return null;
     }
 
+    /**
+     * @param $className
```
Comment (vladis84) [inline framework/base/Module.php]: @param $className string

File: tests/framework/base/ModuleTest.php [hunk 3]
```diff
@@ -120,6 +120,33 @@ public function testServiceLocatorTraversal()
         $this->assertTrue($child->has('test'));
         $this->assertFalse($parent->has('test'));
     }
+
+    public function testCreateControllerByID()
```
Comment (vladis84) [inline tests/framework/base/ModuleTest.php]: I recommend in one test one condition.

File: framework/base/Module.php (hunk 1 above)
Comment (developeruz) [reply framework/base/Module.php]: No, it is part of route and Yii2 splits CamelCase to dashed names.

File: framework/base/Module.php [hunk 4]
```diff
@@ -626,14 +626,13 @@ public function createControllerByID($id)
             $className = substr($id, $pos + 1);
         }
 
-        if (!preg_match('%^[a-z][a-z0-9\\-_]*$%', $className)) {
-            return null;
-        }
-        if ($prefix !== '' && !preg_match('%^[a-z0-9_/]+$%i', $prefix)) {
+        if($this->incorrectClassNameOrPrefix($className, $prefix)) {
```
Comment (samdark) [inline framework/base/Module.php]: `isIncorrectClassNameOrPrefix` is a better name. Also please fix formatting: `if<space>(`

File: framework/base/Module.php [hunk 5]
```diff
@@ -649,6 +648,23 @@ public function createControllerByID($id)
         return null;
     }
 
+    /**
+     * @param $className
+     * @param $prefix
```
Comment (samdark) [inline framework/base/Module.php]: string

File: framework/base/Module.php (hunk 2 above)
Comment (samdark) [reply framework/base/Module.php]: phpdoc is needed.

File: framework/base/Module.php (hunk 5 above)
Comment (samdark) [reply framework/base/Module.php]: phpdoc is needed.

**Review by samdark — CHANGES_REQUESTED** [review]
Overall looks good. Please fix minor things.

File: framework/base/Module.php (hunk 4 above)
Comment (developeruz) [reply framework/base/Module.php]: Done

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**